pkgs/development/python-modules/python-youtube/default.nix

```diff
@@ -0,0 +1,78 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, poetry-core
+, dataclasses-json
+, isodate
+, requests
+, requests-oauthlib
+, pytestCheckHook
+, responses
+, pytest
+, pytest-cov
+,
+}:
+buildPythonPackage rec {
+  pname = "python-youtube";
+  version = "0.9.0";
+  format = "pyproject";
+
+  src = fetchFromGitHub {
+    owner = "sns-sdks";
+    repo = "python-youtube";
+    rev = "v${version}";
+    hash = "sha256-uimipYgf8nfYd1J/K6CStbzIkQiRSosu7/yOfgXYCks=";
+  };
+
+  postPatch = ''
+    substituteInPlace pyproject.toml \
+      --replace "poetry.masonry.api" "poetry.core.masonry.api"
+  '';
+
+  nativeBuildInputs = [
+    poetry-core
+  ];
+
+  propagatedBuildInputs = [
+    dataclasses-json
+    isodate
+    requests
+    requests-oauthlib
+  ];
+
+  pythonImportsCheck = [ "pyyoutube" ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
+    responses
+    pytest
+    pytest-cov
```

Comment:
```suggestion
```
We don't want collect coverage data

Reply:
Do you have an advise on how to disable that flag as it is set in [`pytest.ini` upstream](https://github.com/sns-sdks/python-youtube/blob/master/pytest.ini)?

```console
       > ERROR: usage: __main__.py [options] [file_or_dir] [file_or_dir] [...]
       > __main__.py: error: unrecognized arguments: --cov=pyyoutube --cov-report
       >   inifile: /build/source/pytest.ini
       >   rootdir: /build/source
       > /nix/store/blpvf60m29q02c0lc5fyhim30ma4y1vv-stdenv-linux/setup: line 1596: pop_var_context: head of shell_variables not a function context
```

Reply:
Just remove them with substituteInPlace or sed

Reply:
Thx! Done in [force-push](https://github.com/NixOS/nixpkgs/compare/c7ad518b6de3b2652caf335e94e8b8d8188c1931..1ea6bd071f9f4e108f5b8f3504d18bf7ece79e76).